gen2-yolo/yolo-segmentation/main.py

```diff
@@ -0,0 +1,260 @@
+import depthai as dai
+import numpy as np
+import argparse
+import time
+import random
+
+
+def sigmoid(z):
+  return 1 / (1 + np.exp(-z))
+
+def generate_random_color():
+  return [random.randint(0, 255) for _ in range(3)]
+
+
+coco_classes = ['person', 'bicycle', 'car', 'motorcycle', 'airplane',
+                'bus', 'train', 'truck', 'boat', 'traffic light',
+                'fire hydrant', 'stop sign', 'parking meter', 'bench',
+                'bird', 'cat', 'dog', 'horse', 'sheep', 'cow', 'elephant',
+                'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag',
+                'tie', 'suitcase', 'frisbee', 'skis', 'snowboard', 'sports ball',
+                'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
+                'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife',
+                'spoon', 'bowl', 'banana', 'apple', 'sandwich', 'orange', 'broccoli',
+                'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
+                'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop',
+                'mouse', 'remote', 'keyboard', 'cell phone', 'microwave', 'oven',
+                'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase',
+                'scissors', 'teddy bear', 'hair drier', 'toothbrush']
+
+class_colors = {cls: generate_random_color() for cls in coco_classes}
+
+pixel_conf_threshold = 0.3
+available_yolo_versions = [5, 8, 9, 11]
+
+
+if __name__ == '__main__':
+
+  parser = argparse.ArgumentParser(description="Object segmentation on OAK devices using YOLO models.", formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+  parser.add_argument("--blob", help="Path to YOLO blob file for inference.", required=True, type=str)
+  parser.add_argument("--conf", help="Set the confidence threshold.", default=0.3, type=float)
+  parser.add_argument("--iou", help="Set the NMS IoU threshold.", default=0.5, type=float)
+  parser.add_argument("--version", help=f"Set the YOLO version to consider. Available: {available_yolo_versions}.", required=True, type=int)
+  args = parser.parse_args()
+
+  nn_blob_path = args.blob
+  confidence_threshold = args.conf
+  iou_threshold = args.iou
+  yolo_version = args.version
+
+  if yolo_version not in available_yolo_versions:
+    print(f"Error: You must select a valid YOLO version: {available_yolo_versions}")
+    sys.exit(-1)
+
+  # Start defining a pipeline
+  pipeline = dai.Pipeline()
+  pipeline.setOpenVINOVersion(version = dai.OpenVINO.VERSION_2022_1)
+
+  # Define a neural network that will make predictions based on the source frames
+  nn_node = pipeline.create(dai.node.NeuralNetwork)
+  nn_node.setNumPoolFrames(4)
+  nn_node.input.setBlocking(False)
+  nn_node.setNumInferenceThreads(2)
+  nn_node.setBlobPath(nn_blob_path)
+
+  # Define camera source
+  rgb_cam_node = pipeline.create(dai.node.ColorCamera)
+  rgb_cam_node.setBoardSocket(dai.CameraBoardSocket.CAM_A)
+  rgb_cam_node.setFps(30)
```

Comment:
Could we add `fps` as an optional argument to the ArgumentParser, with default value 30 and use that instead?